### ISSUE TITLE
Barcode implementation

### ISSUE THREAD

**Issue by kcasarez** [RamyTalal/Label-Printer#8]
I'm working on a (partial, for now) implementation of barcode printing based on your other function implementations,, but it's not actually printing a barcode when I try to use it.  Any chance you could look it over and see what I might be missing?

[BarCode.php.txt](https://github.com/RamyTalal/Label-Printer/files/3203163/BarCode.php.txt)

In my PHP page that prints the label, this is the line I use to write the barcode:
`$printer->addCommand(new Command\BarCode($detailID,48,Command\BarCode::WIDTH_MEDIUM,"code39",false,2.5));`

**Comment by RamyTalal** [comment]
Hello,

A (long) while ago I also tried to get barcodes to work, I made this: https://github.com/RamyTalal/Label-Printer/pull/5/files#diff-1f38cad15689411667485a54ba9d2025 but I also can't get it to work.

**Comment by kcasarez** [comment]
I got Code39 barcode working for me.  I took a look at your implementation, and with just changing the preference order of 3 options (basically moving Parenthesis ahead of Height and shifting Height and Width down by 1) I have it working for me with your code too.  I added suggestions on the pull request showing the change.

My only further comment would be that modifying the options could be tough for an end user, because they may not be able to read the code well enough to figure out which order preference they should be modifying when it's just a number.  Since there's a static number of options, and they seem to need to go in a specific order, it seems more accessible to use alphabetic array keys instead of numeric keys, and then have the parseOptions function read them in the specific order they're needed.  That would make input of an options array at construction easier to create. For example:
```
$printer = new Printer(new Escp($stream));
$options = array("height"=>80,"parenthesis"=>Command\BarCode\Parenthesis::OFF,"ratio"=>Command\BarCode\Ratio::TWO_BY_ONE);
$printer->addCommand(new Command\BarCode("123456",Command\BarCode\Format::CODE39,$options));
$printer->printLabel();
```
